packages/babel-helpers/src/helpers.js

```diff
@@ -992,9 +991,18 @@ helpers.maybeArrayLike = helper("7.9.0")`
   }
 `;
 
+helpers.extractIteratorMethod = helper("7.0.0-beta.0")`
+  export default function _extractIteratorMethod(iter) {
+    if ("_es6-shim iterator_" in Object(iter)) return iter["_es6-shim iterator_"];
+    if (typeof Symbol !== "undefined" && Symbol.iterator in Object(iter))
+      return iter[Symbol.iterator];
+  }
```

Comment:
I wrote https://www.npmjs.com/package/es-get-iterator which includes what is actually required to reliably get the iterator method - https://github.com/ljharb/es-get-iterator/blob/master/index.js is surely too large to consider including in babel, but imo that is the correct approach, and it'd be ideal for babel to use it.

If you end up sticking with this simpler approach, be sure to also check for the string "@@iterator" which is used in older Firefoxes.